x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/InstanceTypeService.java

```diff
@@ -27,33 +27,37 @@ private static final class Holder {
         private static final Map<InstanceType, CostEntry> costsPerDatacenter;
 
         static {
```

Comment:
These data sets are loaded when the first request hits a cluster node and consequently this request takes an additional latency hit. I believe we could avoid this by loading data asynchronously but that adds complexity. But as a first step, could you add a `StopWatch` here so we can see how long this step takes?

Reply:
:+1: Added
On my laptop:
```
loadProfilingCostsData took [426.824995 ms]
```

Reply:
Hm, it looks like that removing the pre-allocation shaves off 50-80ms. That is really weird, but I did a couple of tests.

Reply:
Thanks for adding this; that's useful.